README.md

```diff
@@ -559,6 +527,287 @@ You can build a pipeline to use [FHIR converter](https://github.com/microsoft/FH
+Users can list anonymization rules for individual DICOM tag ( by tag value or tag name) as well as a set of tags (by masked value or DICOM VR). Ex：
+```
+{
+    "rules": [
+            {"tag": "(0010,1010)","method": "perturb"}, 
+            {"tag": "(0040,xxxx)",  "method": "redact"},
+            {"tag": "PatientID",  "method": "cryptohash"},
+            {"tag": "PN", "method": "encrypt"}
+    ]
+}
+```
+Parameters in each rules:
+
+|Fields|Description| Valid Value|Required|default value|
+|--|-----|-----|--|--|
+|tag|Used to define DICOM elements |1. Tag Value, e.g. (0010, 0010) or 0010,0010 or 00100010. <br>2. Tag Name. e.g. PatientName. <br> 3. Masked DICOM Tag. e.g. (0010, xxxx) or (xx10, xx10). <br> 4. DICOM VR. e.g. PN, DA.|True|null| 
+|method|De-ID method.| keep, redact, perturb, dateshift, encrypt, cryptohash, substitute, refreshUID, remove.| True|null|
+|setting| Setting for de-id method. Users can add custom settings in the field of "customSettings" and specify setting's name here. |valid setting's name |False|Default setting in the field of "defaultSettings"|
+|params|parameters override setting for de-id methods.|valid parameters|False|null|
+
+Each DICOM tag can only be anonymized once, if two rules have conflicts on one tag, only the former rule will be applied.
+
+### How to set settings
+_defaultSettings_ and _customSettings_ are used to config anonymization method. (Detailed parameters are defined in [anonymization Methods Section]()). _defaultSettings_ are used when user does not specify settings in rule. As for _customSettings_, users need to add the setting with unique name. This setting can be used in "rules" by name.
+
+Here is an example, the first rule will use perturb setting in _defaultSettings_ and the second one will use perturbCustomerSetting in field _cutomSettings_.
+
+```
+{
+    "rules": [
+        {"tag": "(0010,0020)","method": "perturb"},
+        {"tag": "(0010,1010)","method": "perturb", "setting":"perturbCustomerSetting"}
+    ],
+    "defaultSettings":[
+        {"perturb":{ "span": "1", "roundTo": 2, "rangeType": "Proportional", "NoiseFunction" : "Uniform"}}
+    ],
+    "cutomizedSettings":[
+        {"perturbCustomerSetting":{ "span": "1", "roundTo": 2, "rangeType": "Proportional", "NoiseFunction" : "Uniform"}}
+    ]
+}
+```
+
+## Anonymization algorithms
+
+### Overview
+
+
+|De-id Method|Description|Setting Configuration|
+|-----|-----|-----|
+|keep|Retains the value as is.|No|
+|redact|Clean the value.|Yes|
+|remove|Remove the element. |No|
+|perturb|Perturb the value with random noise addition.|Yes|
+|dateShift|Shift the value using the Date-shift algorithm.|Yes|
+|cryptoHash|Transform the value using Crypto-hash method.|Yes|
+|encrypt|Transform the value using Encrypt method.|Yes|
+|substitute|Substitute the value to a predefined value.|Yes|
+|refreshUID|replace with a non-zero length UID|No|
+
+Setting configuration indicates the algorithm needs _defaultSettings_ and _customSettings_ or not.
```

Comment:
this is an illustration for table's header "setting configuration". Do we have better method to add it?